nixos/modules/services/cluster/k3s/default.nix

```diff
@@ -359,6 +359,53 @@ in
         by the k3s agent. This option only makes sense on nodes with an enabled agent.
       '';
     };
+
+    gracefulNodeShutdown = {
+      enable = lib.mkEnableOption ''
+        graceful node shutdowns where the kubelet attempts to detect
+        node system shutdown and terminates pods running on the node. See the
+        [documentation](https://kubernetes.io/docs/concepts/cluster-administration/node-shutdown/#graceful-node-shutdown)
+        for further information.
+      '';
+
+      shutdownGracePeriod = lib.mkOption {
+        type = lib.types.nonEmptyStr;
```

Comment:
I think, we should use `int` here. (Not that I care that much. But a person very much into typing made me re-do after merged [!!] (reverted!) another module because of this.)
It's important to mention the unit will be seconds then.

Reply:
The kubelet uses the [meta/v1.Duration](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#Duration) type, which marshals into duration strings. You could also set it to `1m`, `1m30s`, `1h5m` and so on, so I think a string is more transparent.

Reply:
Well. Let's leave as it is then. At least, I tried. ;-)